# **Task 1: Building a Model to Predict Whether passenger on the Titanic survived or not.**

### **Importing Libraries**

In [ ]:
import numpy  as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, precision_score, recall_score


### **Importing Titanic dataset**

In [ ]:
data=pd.read_csv("/content/Titanic-Dataset.csv")

### **Data Preprocessing**

Drobpping all unwanted columns from the dataset

In [ ]:
df=data.drop(["PassengerId","Name","Ticket","SibSp","Parch","Embarked","Cabin"],axis=1)
df.head()

Survived  Pclass     Sex   Age     Fare
0         0       3    male  22.0   7.2500
1         1       1  female  38.0  71.2833
2         1       3  female  26.0   7.9250
3         1       1  female  35.0  53.1000
4         0       3    male  35.0   8.0500

In [ ]:
missing_values = df.isnull().sum()
missing_values

Survived      0
Pclass        0
Sex           0
Age         177
Fare          0
dtype: int64

*There are 177 missing values in the age*

In [ ]:
df["Age"].fillna(df["Age"].median(), inplace=True)
df['Sex'] = df['Sex'].map({'male': 0, 'female': 1})



<ipython-input-194-da37df6b6a45>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["Age"].fillna(df["Age"].median(), inplace=True)


*Filled the blank ages with median and mapped male ino 0 and female into 1*

In [ ]:
df.head()

Survived  Pclass  Sex   Age     Fare
0         0       3    0  22.0   7.2500
1         1       1    1  38.0  71.2833
2         1       3    1  26.0   7.9250
3         1       1    1  35.0  53.1000
4         0       3    0  35.0   8.0500

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Sex       891 non-null    int64  
 3   Age       891 non-null    float64
 4   Fare      891 non-null    float64
dtypes: float64(2), int64(3)
memory usage: 34.9 KB


### **Exploratory Data Analysis**

In [ ]:
df.describe()

Survived      Pclass         Sex         Age        Fare
count  891.000000  891.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642    0.352413   29.361582   32.204208
std      0.486592    0.836071    0.477990   13.019697   49.693429
min      0.000000    1.000000    0.000000    0.420000    0.000000
25%      0.000000    2.000000    0.000000   22.000000    7.910400
50%      0.000000    3.000000    0.000000   28.000000   14.454200
75%      1.000000    3.000000    1.000000   35.000000   31.000000
max      1.000000    3.000000    1.000000   80.000000  512.329200

*These are the summary measures of the parameters in r the dataset*

In [ ]:
Survived = df['Survived'].value_counts()
Survived


Survived
0    549
1    342
Name: count, dtype: int64

*In this dataset,  0 and 1 representing the non survivers and survivers in the titanic. Most of them are non survivers(569), only few of them are survived (342)*

### **Creating a Model**

In [ ]:
X = df.drop("Survived", axis=1)
y = df["Survived"]
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [ ]:
model = LogisticRegression()
model.fit(X_train, y_train)

LogisticRegression()

In [ ]:
y_pred = model.predict(X_test)
print(f'Accuracy: {accuracy_score(y_test, y_pred)}')
print(f'Precision: {precision_score(y_test, y_pred)}')
print(f'Recall: {recall_score(y_test, y_pred)}')

Accuracy: 0.8044692737430168
Precision: 0.782608695652174
Recall: 0.7297297297297297


###**Interpretaion of Result**

*1. An accuracy of 0.8045 means that the model correctly predicted whether a passenger survived or not about 80.45% of the time.*

*2. precision of 0.7826 means that when the model predicts a passenger survived, it is correct 78.26% of the time.*

*3.recall of 0.7297 means that the model correctly identified 72.97% of the actual survivors.*

### **Conclusion**

*This model achieved an accuracy of approximately 80.45%, precision of 78.26%, and recall of 72.97%. So, we can say the model is a good model.*